### ISSUE TITLE
Fatal failure in nl::detect_ran_encoding() at src/theory/arith/nl/poly_conversion.cpp:668

### ISSUE THREAD

**Issue by mpreiner** [cvc5/cvc5-projects#402]
cvc5/cvc5@acba73710
murxla/murxla@4f03339
```
(set-logic QF_ANRAT)
(declare-const x5 Bool)
(declare-const x (Array Real Real))
(declare-const x8 Real)
(assert (or (not x5) (= 1.0 (select (store x (- (/ (+ 0.0 real.pi) x8)) 0.0) 0.0))))
(check-sat)
```

error:

```
Fatal failure within std::tuple<cvc5::NodeTemplate<true>, cvc5::Rational, cvc5::Rational> cvc5::theory::arith::nl::detect_ran_encoding(const Node&) at src/theory/arith/nl/poly_conversion.cpp:668
Check failure

 n.getNumChildren() == 3
Invalid node structure.
```

**Comment by nafur** [comment]
We will soon use proper `RealAlgebraicNumber` nodes, making the incorrect code obsolete.

**Comment by nafur** [comment]
It will probably fail afterwards anyway, as using transcendental with other theory most likely breaks.

**Comment by nafur** [comment]
This issue was resolved by cvc5/cvc5#7939 and cvc5/cvc5#8076.
